src/middleware/issueDetails.middleware.js

```diff
@@ -228,16 +134,22 @@ export function prepareIssueDetailsTemplateParams (req, res, next) {
     geometries
   }
 
-  const paginationObj = {}
+  const paginationObj = {
+    items: []
+  }
+
+  const entryNumbers = Object.keys(issuesByEntryNumber)
+  const pageNumber = entryNumbers.findIndex(currentEntryNumber => currentEntryNumber === entryNumber) + 1
+
   if (pageNumber > 1) {
     paginationObj.previous = {
-      href: `${BaseSubpath}${pageNumber - 1}`
+      href: `${BaseSubpath}${entryNumbers[pageNumber - 1]}`
     }
   }
 
   if (pageNumber < issueEntitiesCount) {
     paginationObj.next = {
-      href: `${BaseSubpath}${pageNumber + 1}`
+      href: `${BaseSubpath}${entryNumbers[pageNumber + 1]}`
```

Comment:
It's hard to figure out, just by looking at the code, what is the relationship between `entryNumbers`, `issueEntitiesCount` and `pageNumber` (in particular, whether `entryNumbers[pageNumber + 1]` is out of range), so it would be nice to have that stated somewhere.

Also, we are not guaranteed to find the index on line 142 (`entryNumber` comes from request params, so it can be anything)

Reply:
changed this so the url param is now pageNumber instead of entity number, and added some additional checks to make sure that the page Number url param is correct. returning a 404 if the pageNumber doesn't map to an entity